themes/bootstrap3/templates/search/searchbox.phtml

```diff
@@ -81,135 +99,137 @@
         <?php endforeach; ?>
-        <?php if ($hasDefaultsApplied ?? false): ?>
-          <input class="applied-filter" id="dfApplied" type="hidden" name="dfApplied" value="1">
+      <?php endif; ?>
+      <?php if (($hasDefaultsApplied ?? false) || !empty($filterDetails)): ?>
+        <?php if ($options->getRetainFilterSetting()): ?>
+          <?php foreach ($filterDetails as $current): ?>
+            <input class="applied-filter" id="<?=$this->escapeHtmlAttr($current['id'])?>" type="hidden" name="filter[]" value="<?=$this->escapeHtmlAttr($current['value'])?>">
+          <?php endforeach; ?>
+          <?php if ($hasDefaultsApplied ?? false): ?>
+            <input class="applied-filter" id="dfApplied" type="hidden" name="dfApplied" value="1">
+          <?php endif; ?>
         <?php endif; ?>
       <?php endif; ?>
-    <?php endif; ?>
-    <?php foreach ($hiddenFilters as $key => $filter): ?>
-      <?php foreach ($filter as $value): ?>
-        <input type="hidden" name="hiddenFilters[]" value="<?=$this->escapeHtmlAttr($key) . ':' . $this->escapeHtmlAttr('"' . $value . '"')?>">
+      <?php foreach ($hiddenFilters as $key => $filter): ?>
+        <?php foreach ($filter as $value): ?>
+          <input type="hidden" name="hiddenFilters[]" value="<?=$this->escapeHtmlAttr($key) . ':' . $this->escapeHtmlAttr('"' . $value . '"')?>">
+        <?php endforeach; ?>
       <?php endforeach; ?>
-    <?php endforeach; ?>
-    <?php
-      /* Show hidden field for active search class when in combined handler mode. */
-      if ($this->searchbox()->combinedHandlersActive()) {
-        echo '<input type="hidden" name="activeSearchClassId" value="' . $this->escapeHtmlAttr($this->searchClassId) . '">';
-      }
-      /* Load hidden limit preference from Session */
-      if (!empty($lastLimit)) {
-        echo '<input type="hidden" name="limit" value="' . $this->escapeHtmlAttr($lastLimit) . '">';
-      }
-      if (!empty($lastSort) && $lastSort !== $params?->getDefaultSort()) {
-        echo '<input type="hidden" name="sort" value="' . $this->escapeHtmlAttr($lastSort) . '">';
-      }
-    ?>
+      <?php
+        /* Show hidden field for active search class when in combined handler mode. */
+        if ($this->searchbox()->combinedHandlersActive()) {
+          echo '<input type="hidden" name="activeSearchClassId" value="' . $this->escapeHtmlAttr($this->searchClassId) . '">';
+        }
+        /* Load hidden limit preference from Session */
+        if (!empty($lastLimit)) {
+          echo '<input type="hidden" name="limit" value="' . $this->escapeHtmlAttr($lastLimit) . '">';
+        }
+        if (!empty($lastSort) && $lastSort !== $params?->getDefaultSort()) {
+          echo '<input type="hidden" name="sort" value="' . $this->escapeHtmlAttr($lastSort) . '">';
+        }
+      ?>
+    </div>
```

Comment:
I'm not actually sure how much should be included within the div.  I think everything within it here is inline, and that the filters below are supposed to be below.